File: src/gestorAplicación/modelos/Establecimiento.java
package gestorAplicacion.modelos;

import java.util.ArrayList;
import java.util.Iterator;

public class Establecimiento {

	private int tamanoAlmacenamiento;
	private int minProducto;
	private ArrayList<Producto> productos = new ArrayList<Producto>();

	public Establecimiento() {
	}

	public Establecimiento(int tamanoAlmacenamiento, int minProducto) {
		this.tamanoAlmacenamiento = tamanoAlmacenamiento;
		this.minProducto = minProducto;
	}

	public int verDisponibilidad() {
		return 1;
	}

	public int getTamanoAlmacenamiento() {
		return tamanoAlmacenamiento;
	}

	public void setTamanoAlmacenamiento(int tamanoAlmacenamiento) {
		this.tamanoAlmacenamiento = tamanoAlmacenamiento;
	}

	public ArrayList<Producto> getProductos() {
		return productos;
	}

	public void setProductos(Producto producto) {
		this.productos.add(producto);
	}

	public int getMinProducto() {
		return minProducto;
	}

	public void setMinProducto(int minProducto) {
		this.minProducto = minProducto;
	}

	public Producto buscarProductoPorID(int id) {
		Iterator<Producto> itProd = productos.iterator();
		while (itProd.hasNext()) {
			Producto tempProd = itProd.next();
			if (tempProd.getId() == id) {
				try {
					return (Producto) tempProd.clone();
				} catch (CloneNotSupportedException e) {
					e.printStackTrace();
				}
			}
		}
		return null;
	}

	public Producto buscarProductoPorNombre(String nombre) {
		Iterator<Producto> itProd = productos.iterator();
		while (itProd.hasNext()) {
			Producto tempProd = itProd.next();
			if (tempProd.getNombre().equals(nombre)) {
				try {
					return (Producto) tempProd.clone();
				} catch (CloneNotSupportedException e) {
					e.printStackTrace();
				}
			}
		}
		return null;
	}
}

File: python/gestorAplicacion/modelos/Establecimiento.py
import copy

class Establecimiento:
    
    # Constructor
    def __init__(self, tamanoAlmacenamiento=2000, minProducto=10):
        self._tamanoAlmacenamiento = tamanoAlmacenamiento
        self._minProducto = minProducto
        self._productos = []

    # Getter y Setter
    def getTamanoAlmacenamiento(self):
        return self._tamanoAlmacenamiento

    def setTamanoAlmacenamiento(self, tamanoAlmacenamiento):
        self._tamanoAlmacenamiento = tamanoAlmacenamiento

    def getMinProducto(self):
        return self._minProducto

    def setMinProducto(self, minProducto):
        self._minProducto = minProducto

    def getProductos(self):
        return self._productos

    def setProductos(self, productos):
        self._productos = productos


    # Metodos
    def verDisponibilidad(self):
        cantidadOcupada = 0

        for producto in self._productos:
            cantidadOcupada += producto.getCantidad()

        return self._tamanoAlmacenamiento - cantidadOcupada

    def agregarProducto(self, producto):
        self._productos.append(producto)

    def buscarProductoPorIDCopia(self, id):
        for i in self._productos:
            if i.getId() == id:
                return copy.deepcopy(i)
        return None

    def buscarProductoPorIDOriginal(self, id):
        for i in self._productos:
            if i.getId() == id:
                return i

        return None

    def buscarProductoPorNombreCopia(self, nombre):
        for i in self._productos:
            if i.getNombre() == nombre:
                return copy.deepcopy(i)

        return None

    def buscarProductoPorNombreOriginal(self, nombre):
        for i in self._productos:
            if i.getNombre() == nombre:
                return i

        return None
